src/sonic-pit/pit-sysdiag/src/function.py

```diff
@@ -0,0 +1,104 @@
+import subprocess
+import imp
+import json
+import requests
+
+
+PLATFORM_ROOT_PATH = "/usr/share/sonic/device"
+PLATFORM_ROOT_PATH_DOCKER = "/usr/share/sonic/platform"
+
+SONIC_CFGGEN_PATH = "/usr/local/bin/sonic-cfggen"
+HWSKU_KEY = "DEVICE_METADATA['localhost']['hwsku']"
+PLATFORM_KEY = "DEVICE_METADATA['localhost']['platform']"
+
+
+def get_platform_and_hwsku():
+    try:
+        proc = subprocess.Popen([SONIC_CFGGEN_PATH, '-H', '-v', PLATFORM_KEY],
+                                stdout=subprocess.PIPE,
+                                shell=False,
+                                stderr=subprocess.STDOUT)
+        stdout = proc.communicate()[0].decode()
+        proc.wait()
+        platform = stdout.rstrip("\n")
+        # platform = platform.lower().rstrip("-r0")
+
+        proc = subprocess.Popen([SONIC_CFGGEN_PATH, '-d', '-v', HWSKU_KEY],
+                                stdout=subprocess.PIPE,
+                                shell=False,
+                                stderr=subprocess.STDOUT)
+        stdout = proc.communicate()[0].decode()
+        proc.wait()
+        hwsku = stdout.rstrip("\n")
+    except OSError as e:
+        raise OSError("Cannot detect platform")
+
+    return (platform, hwsku)
+
+
+def load_platform_util_module(module_name):
+    platform_util_module = None
+
+    (platform, hwsku) = get_platform_and_hwsku()
+    platform_path = ''
+    if len(platform) != 0:
+        platform_path = "/".join([PLATFORM_ROOT_PATH, platform])
+    else:
+        platform_path = PLATFORM_ROOT_PATH_DOCKER
+
+    try:
+        module_file = "/".join([platform_path, "plugins", module_name + ".py"])
+        platform_util_module = imp.load_source(module_name, module_file)
+    except IOError as e:
+        return None
+
+    return platform_util_module
+
+
+def convert_unicode(input):
+    if isinstance(input, dict):
+        return {convert_unicode(key): convert_unicode(val) for key, val in input.iteritems()}
+    elif isinstance(input, unicode):
+        return input.encode("utf-8")
+    else:
+        return input
+
+
+def exec_cmd(cmd_str):
+    output = None
+    try:
+        output = subprocess.check_output(cmd_str, shell=True,)
+    except Exception as err:
+        print("{} exec cmd failed.{}".format(cmd_str, err))
+    return output
+
+
+def run_command(cmd):
```

Comment:
Can "exec_cmd" and "run_command" be merged into one function?

Reply:
Remove function exec_cmd(cmd_str), as it can be replaced by function run_command(cmd).